collector/src/bin/rustc-fake.rs

```diff
@@ -37,6 +37,20 @@ fn run_with_determinism_env(mut cmd: Command) {
     );
 }
 
+// We want each rustc execution to have a separate self-profile directory,
+// to avoid overwriting the results. PID of this process and timestamp should
+// hopefully be unique enough.
```

Comment:
I think just the PID should be unique?

Reply:
Linux can recycle PIDs, so in theory another rustc-fake running later could have the same PID.